app/src/main/java/vn/mbm/phimp/me/gallery/GalleryImageAdapter.java

```diff
@@ -24,214 +25,221 @@
+        mFilePath = filepath;
+
+    }
+
+
+    public int getCount() {
+        return mFilePath.size();
+    }
+
+    public void onChange() {
+        mFilePath.remove(PhimpMeGallery.position);
+        notifyDataSetChanged();
+    }
+
+    public Object getItem(int position) {
+        return null;
+    }
+
+
+    public long getItemId(int position) {
+        return 0;
+    }
+
+
+    public View getView(int position, View convertView, ViewGroup parent) {
+        PhimpMeGallery.position = position;
+        if (convertView == null) {
+
+            holder = new ViewHolder();
+
+            imageView = new ImageView(this.context);
+
+            imageView.setPadding(3, 3, 3, 3);
+
+            convertView = imageView;
+
+            holder.imageView = imageView;
+
+            convertView.setTag(holder);
+
+        } else {
+
+            holder = (ViewHolder) convertView.getTag();
+        }
+
+
+        try {
+            holder.imageView.setImageBitmap(decodeSampledBitmapFromFile(context, mFilePath.get(position), screen_w));
+        } catch (IndexOutOfBoundsException e) {
+            holder.imageView.setImageDrawable(context.getResources().getDrawable(R.drawable.image_not_found));
+        }
+        //holder.imageView.setImageBitmap(BitmapFactory.decodeResource(context.getResources(),R.drawable.image_not_found, null));
+        //bm = null;
+
+        holder.imageView.setLayoutParams(new Gallery.LayoutParams(Gallery.LayoutParams.MATCH_PARENT, Gallery.LayoutParams.MATCH_PARENT));
+
+        holder.imageView.setScaleType(ImageView.ScaleType.FIT_CENTER);
+
+        return imageView;
+
+    }
+
+    public static int calculateInSampleSize(
+
+            String mfile, int reqWidth) {
+        // Raw height and width of image
+        //Bitmap bm = null;
+        BitmapFactory.Options bounds = new BitmapFactory.Options();
+        bounds.inTempStorage = new byte[32 * 1024];
+        bounds.inDither = false;
+        bounds.inPurgeable = true;
+        bounds.inInputShareable = true;
+        bounds.inJustDecodeBounds = true;
+        File file = new File(mfile);
+        float rotate = getOrientation(mfile);
+        FileInputStream fs = null;
+        try {
+            fs = new FileInputStream(file);
+        } catch (FileNotFoundException e) {
+            //TODO do something intelligent
+            e.printStackTrace();
+        }
+
+        try {
+            if (fs != null) BitmapFactory.decodeFileDescriptor(fs.getFD(), null, bounds);
+        } catch (IOException e) {
+            //TODO do something intelligent
+            e.printStackTrace();
+        } finally {
+            if (fs != null) {
+                try {
+                    fs.close();
+                } catch (IOException e) {
+                    // TODO Auto-generated catch block
+                    e.printStackTrace();
+                }
+            }
+        }
+        //final int height = bounds.outHeight;
+        final int width;
+        if (rotate == 0)
```

Comment:
@duchuyctlk I managed the feature to add the functionality. These lines must have changed due to auto indentation. The code was written using earlier methods, when some features were not present, this must be to deal with the device rotation.
We are trying to add features for now, thanks.